src/sentry/static/sentry/app/components/bases/pluginComponentBase.tsx

```diff
@@ -11,17 +11,21 @@ import {
 } from 'app/actionCreators/indicator';
 import {t} from 'app/locale';
 
-const callbackWithArgs = function (callback, ...args) {
-  if (isFunction(callback)) {
-    callback = callback.bind(this, ...args);
-  } else {
-    callback = null;
-  }
-  return callback;
+const callbackWithArgs = function (context: any, callback: any, ...args: any) {
+  return isFunction(callback) ? callback.bind(context, ...args) : null;
 };
 
-class PluginComponentBase extends React.Component {
-  constructor(props, context) {
+type GenericFieldProps = Parameters<typeof GenericField>[0];
+
+type Props = {};
```

Comment:
don't you need the `config` prop here anymore?

Reply:
@priscilawebdev Nah, it's always going to be passed to `renderField` and doesn't need a default value.